### ISSUE TITLE
Boolean CLI options in standard "--someFlag=someValue" form fail with "error TS5023: Unknown compiler option..."

### ISSUE THREAD

**Issue by broofa** [microsoft/TypeScript#52655]
# Bug Report

tl;dr: User's expect to assign CLI flag options with an equal sign.  E.g.`--flag=value`.  That this fails in `tsc` with an unobvious error is confusing.  It misleads users into thinking there isn't a way of disabling options from the command line. [For example](https://github.com/Microsoft/TypeScript/issues/30661#issuecomment-629780928):
>  Even if you wanted to forgo the speed benefits of "incremental"... you can't turn it off temporarily because there is no --noIncremental (or analogous) flag.

The form above is wide-spread, especially within the JS community ...

* [node's command line options](https://nodejs.org/api/cli.html#options)
* [commander.js options](https://github.com/tj/commander.js#options)
* [yargs options](https://github.com/yargs/yargs#description)
* [minimist options](https://github.com/minimistjs/minimist#example)

### 🔎 Search Terms

`error TS5023: Unknown compiler option`

### 🕗 Version & Regression Information

Boolean CLI options were first introduced in #11798 , but have never supported the standard `--flag=value` format

* [ ] This is a crash
* [ ] This changed between versions ______ and _______
* [x] This is the behavior in every version I tried, and I reviewed the FAQ for entries about _________
* [ ] I was unable to test this on prior versions because _______

### 🙁 Actual behavior

Providing boolean CLI options  like `--incremental=false`, `--noEmit=false`, and so on all result in a `TS5023: Unknown compiler option` error.

### 🙂 Expected behavior

Using the standard double-dash notation of "--flagName=flagValue" (.e.g `--incremental=false`) should work as expected.
